app/src/main/kotlin/com/mapzen/erasermap/controller/MainActivity.kt

```diff
@@ -711,11 +711,15 @@ class MainActivity : AppCompatActivity(), MainViewController,
     }
 
     override fun addSearchResultsToMap(features: List<Feature>?, activeIndex: Int) {
-        if (features == null) {
+        if (features == null || features.size == 0) {
```

Comment:
Moving these checks into the presenter is on my radar/"next to refactor" list. I didn't do it here because I didn't want to add more changes to this PR but I completely agree that they should be moved into the presenter and covered with tests.

The reason we have these two methods is because we need one method to adjust the views on the screen when search results should be shown (make the search results view visible, adjust the attribution & find me btn)...thats `MainActivity.showSearchResults(features: List<Feature>?)`. And then we have a more specific method which handles updating the map's search result pins/position/zoom (`MainActivity.addSearchResultsToMap(features: List<Feature>?, activeIndex: Int)`) when the search results pager is adjusted.

When search results are first shown, or when we are backing out of route preview, we call `showSearchResults` so that hidden/moved views will be properly adjusted and search results on the map will be shown. But, when the user pages the search results, we don't need to update all the other views on the screen, only remove/re-add the search result pins so that the newly select result is "active"

I think that the logic in `showSearchResults` should  be moved to the presenter so that it can handle calling both and we can add test coverage for this.

I think there wouldn't be any harm in combining them into a single method but I would lean towards keeping them separate and documenting them. This way, we aren't making unnecessary calls to update ancillary views when only the active search index changes. I think `addSearchResultsToMap` is clear but maybe `showSearchResults` could  become `prepareAndShowSearchResultsViews`.

I will be addressing this in the next batch of EM refactoring.